### PR TITLE
[Doc] Add Chinese doc for tutorials01_config_md

### PR DESCRIPTION
**Pull request by MengzhangLI**
Add Chinese doc for tutorials01_config_md

### CODE REVIEW

File: docs_zh-CN/tutorials/config.md [hunk 1]
```diff
@@ -1 +1,377 @@
 # 教程 1: 学习配置文件
+
+我们整合了模块和继承设计到我们的配置里，这便于做很多实验。如果您像检查配置文件，您可以运行 `python tools/print_config.py /PATH/TO/CONFIG` 去查看完整的配置文件。您还可以传递参数
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:3]:
```suggestion
我们整合了模块和继承设计到我们的配置里，这便于做很多实验。如果您想查看配置文件，您可以运行 `python tools/print_config.py /PATH/TO/CONFIG` 去查看完整的配置文件。您还可以传递参数
```

File: docs_zh-CN/tutorials/config.md [hunk 2]
```diff
@@ -1 +1,377 @@
...
+`--options xxx.yyy=zzz` 去查看更新的配置。
+
+## 配置文件的结构
+
+在 `config/_base_` 下面有4个基本构成成分(component)： 数据集(dataset)，模型(model)，计划表(schedule)和默认运行时间(default runtime)。
+这样，像 DeepLabV3, PSPNet 这样的模型可以容易地被构造。被来自 `_base_` 的构成成分所组成的配置文件叫做 `_primitive_`。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:9]:
```suggestion
这样，像 DeepLabV3, PSPNet 这样的模型可以容易地被构造。被来自 `_base_` 的构成成分所组成的配置文件叫做 _primitive_。
```

File: docs_zh-CN/tutorials/config.md [hunk 3]
```diff
@@ -1 +1,377 @@
...
+为了帮助使用者熟悉这个流行的语义分割框架的完整配置文件和模块，我们在下面对使用 ResNet50V1c 的 PSPNet 的配置文件做了详细的注释说明。
+更多的详细使用和其他模块的替代项请参考 API 文档。
+
+```python
+norm_cfg = dict(type='SyncBN', requires_grad=True)  # 分割框架通常使用 SyncBN
+model = dict(
+    type='EncoderDecoder',  # 分割器(segmentor)的名字
+    pretrained='open-mmlab://resnet50_v1c',  # 将被加载的 ImageNet 预训练骨架
+    backbone=dict(
+        type='ResNetV1c',  # 骨架的类别。 可用选项请参考 mmseg/backbone/resnet.py
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:47]:
```suggestion
        type='ResNetV1c',  # 主干网络的类别。 可用选项请参考 mmseg/backbone/resnet.py
```

File: docs_zh-CN/tutorials/config.md [hunk 4]
```diff
@@ -1 +1,377 @@
...
+## PSPNet 的一个例子
+
+为了帮助使用者熟悉这个流行的语义分割框架的完整配置文件和模块，我们在下面对使用 ResNet50V1c 的 PSPNet 的配置文件做了详细的注释说明。
+更多的详细使用和其他模块的替代项请参考 API 文档。
+
+```python
+norm_cfg = dict(type='SyncBN', requires_grad=True)  # 分割框架通常使用 SyncBN
+model = dict(
+    type='EncoderDecoder',  # 分割器(segmentor)的名字
+    pretrained='open-mmlab://resnet50_v1c',  # 将被加载的 ImageNet 预训练骨架
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:45]:
```suggestion
    pretrained='open-mmlab://resnet50_v1c',  # 将被加载的 ImageNet 预训练主干网络
```

File: docs_zh-CN/tutorials/config.md [hunk 5]
```diff
@@ -1 +1,377 @@
...
+
+我们整合了模块和继承设计到我们的配置里，这便于做很多实验。如果您像检查配置文件，您可以运行 `python tools/print_config.py /PATH/TO/CONFIG` 去查看完整的配置文件。您还可以传递参数
+`--options xxx.yyy=zzz` 去查看更新的配置。
+
+## 配置文件的结构
+
+在 `config/_base_` 下面有4个基本构成成分(component)： 数据集(dataset)，模型(model)，计划表(schedule)和默认运行时间(default runtime)。
+这样，像 DeepLabV3, PSPNet 这样的模型可以容易地被构造。被来自 `_base_` 的构成成分所组成的配置文件叫做 `_primitive_`。
+
+对于所有在同一个文件夹下的配置文件，推荐**只有一个** primitive 配置文件。所有其他的配置文件都应该继承自这个 primitive 配置文件。以这样的方式，最大的继承深度是3。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:11]: We may refer to https://mmaction2.readthedocs.io/zh_CN/latest/tutorials/1_config.html.

File: docs_zh-CN/tutorials/config.md [hunk 6]
```diff
@@ -1 +1,377 @@
...
+            use_sigmoid=False,  # 在分割里是否使用 sigmoid 激活。
+            loss_weight=0.4)))  # 辅助头里损失的权重。默认设置为0.4。
+train_cfg = dict()  # train_cfg 当前仅是一个占位符。
+test_cfg = dict(mode='whole')  # 测试模式， 选项是 'whole' 和 'sliding'. 'whole': 整张图像全卷积(fully-convolutional)测试。 'sliding': 图像上做滑动裁剪窗口(sliding crop window)。
+dataset_type = 'CityscapesDataset'  # 数据集类型，这将被用来定义数据集。
+data_root = 'data/cityscapes/'  # 数据的根路径。
+img_norm_cfg = dict(  # 图像归一化配置，用来归一化输入的图像。
+    mean=[123.675, 116.28, 103.53],  # 预训练里用于预训练骨架模型的平均值。
+    std=[58.395, 57.12, 57.375],  # 预训练里用于预训练骨架模型的标准差。
+    to_rgb=True)  # 预训练里用于预训练骨架模型的图像的通道阶数。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:95]:
```suggestion
    to_rgb=True)  # 预训练里用于预训练主干网络的图像的通道顺序。
```

File: docs_zh-CN/tutorials/config.md [hunk 7]
```diff
@@ -1 +1,377 @@
...
+crop_size = (512, 1024)  # 训练时的裁剪大小
+train_pipeline = [  #训练管道
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:97]:
```suggestion
train_pipeline = [  #训练流程
```

File: docs_zh-CN/tutorials/config.md [hunk 8]
```diff
@@ -1 +1,377 @@
...
+    dict(type='LoadImageFromFile'),  # 第1个管道，从文件路径里加载图像。
+    dict(type='LoadAnnotations'),  # 第2个管道，对于当前图像，加载它的注释信息。
+    dict(type='Resize',  # 变化图像和其注释大小的数据增广的管道。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:100]:
```suggestion
    dict(type='Resize',  # 变化图像和其注释大小的数据增广的流程。
```

File: docs_zh-CN/tutorials/config.md [hunk 9]
```diff
@@ -1 +1,377 @@
...
+        img_scale=(2048, 1024),  # 图像的最大规模。
+        ratio_range=(0.5, 2.0)), # 数据增广的比例范围。
+    dict(type='RandomCrop',  # 随机裁剪当前图像和其注释大小的数据增广的管道。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:103]:
```suggestion
    dict(type='RandomCrop',  # 随机裁剪当前图像和其注释大小的数据增广的流程。
```

File: docs_zh-CN/tutorials/config.md [hunk 10]
```diff
@@ -1 +1,377 @@
...
+    dict(type='Pad',  # 填充当前图像到指定大小的数据增广的管道。
+        size=(512, 1024),  # 填充的图像大小。
+        pad_val=0,  # 图像的填充值。
+        seg_pad_val=255),  # 'gt_semantic_seg'的填充值。
+    dict(type='DefaultFormatBundle'),  # 管道里收集数据的默认格式捆。
+    dict(type='Collect',  # 决定数据里哪些键被传递到分割器里的管道。
+        keys=['img', 'gt_semantic_seg'])
+]
+test_pipeline = [
+    dict(type='LoadImageFromFile'),  # 第1个管道，从文件路径里加载图像。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:124]:
```suggestion
    dict(type='LoadImageFromFile'),  # 第1个流程，从文件路径里加载图像。
```

File: docs_zh-CN/tutorials/config.md [hunk 11]
```diff
@@ -1 +1,377 @@
...
+    dict(
+        type='MultiScaleFlipAug',  # 封装测试时数据增广(test time augmentations)。
+        img_scale=(2048, 1024),  # 决定测试时可改变图像的最大规模。用于改变图像大小的管道。
+        flip=False,  # 测试时是否翻转图像。
+        transforms=[
+            dict(type='Resize',  # 使用改变图像大小的数据增广。
+                 keep_ratio=True),  # 是否保持宽和高的比例，这里的图像比例设置将覆盖上面的图像规模大小的设置。
+            dict(type='RandomFlip'),  # 考虑到 RandomFlip 已经被添加到管道里，当 flip=False 时它将不被使用。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:132]:
```suggestion
            dict(type='RandomFlip'),  # 考虑到 RandomFlip 已经被添加到流程里，当 flip=False 时它将不被使用。
```

File: docs_zh-CN/tutorials/config.md [hunk 12]
```diff
@@ -1 +1,377 @@
...
+                to_rgb=True),
+            dict(type='ImageToTensor', # 将图像转为张量
+                keys=['img']),
+            dict(type='Collect', # 收集测试时必须的键的收集管道。
+                keys=['img'])
+        ])
+]
+data = dict(
+    samples_per_gpu=2,  # 单个 GPU 的 Batch size
+    workers_per_gpu=2,  # 单个 GPU 分配的线程数
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:146]:
```suggestion
    workers_per_gpu=2,  # 单个 GPU 分配的数据加载线程数
```

File: docs_zh-CN/tutorials/config.md [hunk 13]
```diff
@@ -1 +1,377 @@
...
+    hooks=[
+        # dict(type='TensorboardLoggerHook')  # 同样支持 Tensorboard 日志
+        dict(type='TextLoggerHook', by_epoch=False)
+    ])
+dist_params = dict(backend='nccl')  # 用于设置分布式训练的参数，端口也同样可被设置。
+log_level = 'INFO'  # 日志的级别。
+load_from = None  # 从一个给定路径里加载模型作为预训练模型，它并不会消耗训练时间。
+resume_from = None  # 从给定路径里恢复检查点(checkpoints)，训练模式将从检查点保存的轮次开始恢复训练。
+workflow = [('train', 1)]  # runner 的工作流程。 [('train', 1)] 意思是只有一个工作流程而且工作流程 'train' 仅执行一次。根据 `runner.max_iters` 工作流程训练模型的迭代轮数为40000次。
+cudnn_benchmark = True  # 是否是使用 cudnn_benchmark 去加速，它对于固定输入大小的可以很快加速。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:226]:
```suggestion
cudnn_benchmark = True  # 是否是使用 cudnn_benchmark 去加速，它对于固定输入大小的可以提高训练速度。
```

File: docs_zh-CN/tutorials/config.md [hunk 14]
```diff
@@ -1 +1,377 @@
...
+    policy='poly',  # 调度流程的策略，同样支持 Step, CosineAnnealing, Cyclic 等. 请从 https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/lr_updater.py#L9 参考 LrUpdater 的细节。
+    power=0.9,  # 多项式衰减 (polynomial decay) 的幂。
+    min_lr=0.0001,  # 用来稳定训练的最小学习率。
+    by_epoch=False)  # 是否按照每个 epoch 去算学习率。
+runner = dict(
+    type='IterBasedRunner', # 将使用的 runner 的类别 (例如 IterBasedRunner 或 EpochBasedRunner)。
+    max_iters=40000) # 全部迭代轮数大小，对于 EpochBasedRunner 使用 `max_epochs` 。
+checkpoint_config = dict(  # 设置检查点钩 (checkpoint hook) 的配置文件。执行时请参考 https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/checkpoint.py。
+    by_epoch=False,  # 是否按照每个 epoch 去算 runner。
+    interval=4000)  # 保存的间歇点
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:243]:
```suggestion
    interval=4000)  # 保存的间隔
```

File: docs_zh-CN/tutorials/config.md [hunk 15]
```diff
@@ -1 +1,377 @@
...
+optimizer_config = dict()  # 用于构建优化器钩 (optimizer hook) 的配置文件，执行细节请参考 https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/optimizer.py#L8。
+lr_config = dict(
+    policy='poly',  # 调度流程的策略，同样支持 Step, CosineAnnealing, Cyclic 等. 请从 https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/lr_updater.py#L9 参考 LrUpdater 的细节。
+    power=0.9,  # 多项式衰减 (polynomial decay) 的幂。
+    min_lr=0.0001,  # 用来稳定训练的最小学习率。
+    by_epoch=False)  # 是否按照每个 epoch 去算学习率。
+runner = dict(
+    type='IterBasedRunner', # 将使用的 runner 的类别 (例如 IterBasedRunner 或 EpochBasedRunner)。
+    max_iters=40000) # 全部迭代轮数大小，对于 EpochBasedRunner 使用 `max_epochs` 。
+checkpoint_config = dict(  # 设置检查点钩 (checkpoint hook) 的配置文件。执行时请参考 https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/checkpoint.py。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:241]:
```suggestion
checkpoint_config = dict(  # 设置检查点钩子 (checkpoint hook) 的配置文件。执行时请参考 https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/checkpoint.py。
```

**Review by xvjiarui — COMMENTED** [review]
We may refer to https://mmaction2.readthedocs.io/zh_CN/latest/tutorials/1_config.html for some terminologies.

File: docs_zh-CN/tutorials/config.md (hunk 5 above)
Comment (MengzhangLI) [reply docs_zh-CN/tutorials/config.md:11]: > We may refer to https://mmaction2.readthedocs.io/zh_CN/latest/tutorials/1_config.html.

OK, I will follow and check that doc as template.

File: docs_zh-CN/tutorials/config.md [hunk 16]
```diff
@@ -1 +1,377 @@
...
+我们按照下面的风格去命名配置文件。社区贡献者被建议使用同样的风格。
+
+```
+{model}_{backbone}_[misc]_[gpu x batch_per_gpu]_{resolution}_{schedule}_{dataset}
+```
+
+`{xxx}` 是被要求的文件 `[yyy]` 是可选的。
+
+- `{model}`: 模型种类，例如 `psp`， `deeplabv3` 等等。
+- `{backbone}`: 骨架种类，例如 `r50` (ResNet-50)， `x101` (ResNeXt-101)。
```
Comment (xvjiarui) [inline docs_zh-CN/tutorials/config.md:30]:
```suggestion
- `{backbone}`: 骨干网络种类，例如 `r50` (ResNet-50)， `x101` (ResNeXt-101)。
```

Comment (MengzhangLI) [reply docs_zh-CN/tutorials/config.md:30]: I would replace all "骨架" into "主干网络".

Comment (MengzhangLI) [reply docs_zh-CN/tutorials/config.md:30]: Done.

**Comment by MengzhangLI** [comment]
Just pulled new version, please check it out.

Best,

**Review by Junjun2016 — APPROVED** [review]

**Pull request merged**